src/App.tsx

```diff
@@ -21,55 +20,152 @@ enum SortType {
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
+type State = {
   goods: string[],
-  sortType: SortType,
+  isVisible: boolean,
   isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  isSorted: boolean,
+  sortBy: SortType,
+};
 
-  // Sort and reverse goods if needed
-  // ...
+class App extends React.Component<{}, State> {
+  state:Readonly<State> = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    isSorted: false,
+    sortBy: SortType.NONE,
+  };
 
-  return visibleGoods;
-}
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({
+        goods: goodsFromServer,
+        isVisible: true,
+        isReversed: false,
+        isSorted: false,
+        sortBy: SortType.NONE,
+      });
+    }, 100000000000);
+  }
 
-// DON'T save goods to the state
-type State = {
-  isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
-};
+  visibleGoodsList = () => {
+    this.setState((state) => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  SortByLength = () => {
```

Comment:
use `camelCase` for function names, fix it everywhere
```suggestion
  sortByLength = () => {
```